rust/protocol/src/curve.rs

```diff
@@ -15,8 +19,10 @@ use arrayref::array_ref;
 use rand::{CryptoRng, Rng};
 use subtle::ConstantTimeEq;
 
+/// TODO: describe the intent of this enum!
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
 pub enum KeyType {
+    /// TODO: What does this case mean?
     Djb,
 }
```

Comment:
Currently the only type of key Signal supports is [djb](https://en.wikipedia.org/wiki/Daniel_J._Bernstein) Curve25519 keys, but in theory we could move to another key type that supports signatures and Diffie-Helman agreements in the future (probably another [elliptic curve](https://en.wikipedia.org/wiki/Elliptic-curve_cryptography), but that's not inherently necessary).

Reply:
…I'd be happy to rename "DJB" to "Curve25519" throughout, though, because they really are used synonymously here.

Reply:
Note that this makes it incorrect to expose `PUBLIC_KEY_LENGTH` and `PRIVATE_KEY_LENGTH`; they can be `CURVE25519_PUBLIC_KEY_LENGTH` and `CURVE25519_PRIVATE_KEY_LENGTH` where applicable (or `curve25519::PUBLIC_KEY_LENGTH` and `curve25519::PRIVATE_KEY_LENGTH`), but that's not everywhere in these APIs.

Reply:
I have now simply avoided exposing any of those constants at all outside of `curve.rs` as nothing at this phase is relying on them!

Reply:
There are definitely other places in the larger Signal codebase that want to know (or worse, make assumptions) but we'll deal with that as they come. :sob: